packages/frames.js/src/core/utils.ts

```diff
@@ -24,65 +25,97 @@ function isValidButtonAction(action: any): action is "post" | "post_redirect" {
   );
 }
 
-/**
- * This function generates fully qualified URL for post and post_redirect buttons
- * that also encodes the button type and button value so we can use that on next frame.
- */
-export function generatePostButtonTargetURL({
-  buttonIndex,
-  buttonAction,
+export function generateTargetURL({
   currentURL,
   target,
   basePath,
-  state,
 }: {
-  buttonIndex: 1 | 2 | 3 | 4;
-  buttonAction: "post" | "post_redirect";
   currentURL: URL;
   basePath: string;
-  target: string | undefined;
-  /**
-   * Button state
-   */
-  state: JsonValue | undefined;
+  target: string | UrlObject | undefined;
 }): string {
   let url = new URL(currentURL);
 
-  if (target) {
+  if (
+    target &&
+    typeof target == "string" &&
+    (target.startsWith("http://") || target.startsWith("https://"))
+  ) {
+    // handle absolute urls
+    url = new URL(target);
+  } else if (target && typeof target === "string") {
     // resolve target relatively to basePath
     const baseUrl = new URL(basePath, currentURL);
     const preformatted = baseUrl.pathname + "/" + target;
     const parts = preformatted.split("/").filter(Boolean);
     const finalPathname = parts.join("/");
 
     url = new URL(`/${finalPathname}`, currentURL);
+  } else if (target && typeof target === "object") {
+    // resolve target relatively to basePath
+    url = new URL(
+      formatUrl({
+        ...url,
+        ...target,
+        pathname: [basePath ?? "/", target.pathname].filter(Boolean).join("/"),
```

Comment:
@davidfurlong isn't this generating multiple slashes? for basePath: '/' and `pathname: '/'` it will produce '///` right?

Reply:
good point